File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:

        freq = {}

        for i in range(len(nums)):
            if nums[i] not in freq:
                freq[nums[i]] = 1
            else:
                freq[nums[i]] += 1

        print(freq)

        for key in freq:
            if freq[key] > 1:
                return True
   
        return False

         

File: Data Structures & Algorithms/maximum-subarray/submission-2.go
func maxSubArray(nums []int) int {
    maxSum := nums[0]
    currentSum := 0

    for _, val := range nums{
        if currentSum < 0{
            currentSum = 0
        }
        currentSum += val
        if maxSum < currentSum{
            maxSum = currentSum
        }
    }
    return maxSum
}
